dask_cuda/tests/test_spill.py

```diff
@@ -7,50 +7,63 @@
 from distributed import Client, get_worker, wait
 from dask_cuda import LocalCUDACluster
 from dask_cuda.device_host_file import DeviceHostFile
+from zict.file import _safe_key as safe_key
+
+import dask
 import dask.array as da
 import cupy
-from zict.file import _safe_key as safe_key
+import cudf
 
 
-def assert_device_host_file_size(dhf, total_bytes, chunk_overhead=1024):
+def assert_device_host_file_size(
+    dhf, total_bytes, device_chunk_overhead=0, serialized_chunk_overhead=1024
+):
     byte_sum = dhf.device_buffer.fast.total_weight + dhf.host_buffer.fast.total_weight
     file_path = [os.path.join(dhf.disk.directory, safe_key(k)) for k in dhf.disk.keys()]
     file_size = [os.path.getsize(f) for f in file_path]
     byte_sum += sum(file_size)
 
     # Allow up to chunk_overhead bytes overhead per chunk on disk
-    host_overhead = len(dhf.host) * chunk_overhead
-    disk_overhead = len(dhf.disk) * chunk_overhead
+    device_overhead = len(dhf.device) * device_chunk_overhead
+    host_overhead = len(dhf.host) * serialized_chunk_overhead
+    disk_overhead = len(dhf.disk) * serialized_chunk_overhead
     assert (
         byte_sum >= total_bytes
-        and byte_sum <= total_bytes + host_overhead + disk_overhead
+        and byte_sum <= total_bytes + device_overhead + host_overhead + disk_overhead
+    )
+
+
+def worker_assert(total_size, device_chunk_overhead, serialized_chunk_overhead):
+    assert_device_host_file_size(
+        get_worker().data, total_size, device_chunk_overhead, serialized_chunk_overhead
     )
 
 
 @pytest.mark.parametrize(
     "params",
     [
         {
-            "device_memory_limit": 2e9,
+            "device_memory_limit": 1e9,
             "memory_limit": 4e9,
             "host_target": 0.6,
             "host_spill": 0.7,
             "spills_to_disk": False,
         },
         {
             "device_memory_limit": 1e9,
-            "memory_limit": 1e9,
-            "host_target": 0.3,
-            "host_spill": 0.3,
+            "memory_limit": 2e9,
+            "host_target": 0.6,
+            "host_spill": 0.6,
             "spills_to_disk": True,
         },
     ],
 )
-def test_device_spill(params):
+def test_cupy_device_spill(params):
     @gen_cluster(
         client=True,
         ncores=[("127.0.0.1", 1)],
         Worker=Worker,
+        timeout=300,
```

Comment:
How long does this test usually take?  Five minutes seems like a very long time.  Is this timeout still necessary?

Reply:
I think most of the tests here run in under 60 seconds, but the default isn't enough for some of them, in particular there was one of the CuPy tests that I've seen failing non-deterministically. Since we're not concerned about the timeout in itself for the tests here, IMHO, it's better to have longer timeouts than having some tests failing now and then. In other words, the 300 seconds becomes more of a trigger to avoid CI from hanging forever in an eventual failure, but also prevents it from failing when it just took a while longer due to some unexpected slowness of the CI system.